tests.md

```diff
@@ -0,0 +1,228 @@
+
+### Requirements
+
+It takes little to make the tests. The first thing is to have an infrastructure with the usual configuration of your system. On the other hand, a database where you can test (as it will be created and destroyed).
+
+
+### Installing the test
+
+The first thing we'll do is download and synchronize the tool.
+
+```bash
+cd /tmp/
+git clone https://github.com/WordPress/phpunit-test-runner.git
+cd phpunit-test-runner/
+```
+
+The next step will be to configure the environment. To do this, first we'll make a copy of the example file and then we'll configure it.
+
+```bash
+cp .env.default .env
+vim .env
+```
+
+The content (in summary form) can be something like this:
+
+```bash
+# Path to the directory where files can be prepared before being delivered to the environment.
+export WPT_PREPARE_DIR=/tmp/wordpress
+
+# Path to the directory where the WordPress develop checkout can be placed and tests can be run. When running tests in the same environment, set WPT_TEST_DIR to WPT_PREPARE_DIR
+export WPT_TEST_DIR=/tmp/wordpress
+
+# API key to authenticate with the reporting service in 'username:password' format.
+export WPT_REPORT_API_KEY=
+
+# (Optionally) define an alternate reporting URL
+export WPT_REPORT_URL=
+
+# Credentials for a database that can be written to and reset. WARNING!!! This database will be destroyed between tests. Only use safe database credentials.
+export WPT_DB_NAME=wordpress
+export WPT_DB_USER=wordpress
+export WPT_DB_PASSWORD=__PASSWORD__
+export WPT_DB_HOST=localhost
+
+# (Optionally) set a custom table prefix to permit concurrency against the same database.
+export WPT_TABLE_PREFIX=${WPT_TABLE_PREFIX-wptests_}
+
+# (Optionally) define the PHP executable to be called
+export WPT_PHP_EXECUTABLE=${WPT_PHP_EXECUTABLE-php}
+
+# (Optionally) define the PHPUnit command execution call. Use if `php phpunit.phar` can't be called directly for some reason.
+export WPT_PHPUNIT_CMD=
+
+# (Optionally) define the command execution to remove the test directory. Use if `rm -r` can't be called directly for some reason.
+export WPT_RM_TEST_DIR_CMD=
+
+# SSH connection string (can also be an alias). Leave empty if tests are meant to run in the same environment.
+export WPT_SSH_CONNECT=
+
+# Any options to be passed to the SSH connection. Defaults to '-o StrictHostKeyChecking=no'
+export WPT_SSH_OPTIONS=
+
+# SSH private key, base64 encoded.
+export WPT_SSH_PRIVATE_KEY_BASE64=
+
+# Output logging. Use 'verbose' to increase verbosity
+export WPT_DEBUG=
+```
+
+We will configure the folder where the WordPress software downloads and the database accesses will be made in order to prepare the tests.
+
+### Preparing the environment
+
+Before we do the first test, we'll catch up on everything. This process can be run before each test in this environment if we want to keep it up to date, although it will depend more if it is in a production environment.
+
+```bash
+cd /tmp/phpunit-test-runner/
+git pull
+npm update
```

Comment:
I put it there because there was the "environment preparation", like "here we do all the configurations, installations, and everything, then you only focus on the test". usually (this is personal) when I do documentation like "step-by-step" I prefer to prepare everything if I know I'll need it later. For example, With a new VPS, add the repositories first, update everything... and then, install step by step each software.

Although this tutorial is focused on hosting companies, it is possible that there are also sysadmins, programmers or any other type of profile you want to try and do not have advanced knowledge of systems, so simplify and not mix the steps, in general, has always been a good solution for the courses I have given to other people.

Reply:
I guess I mean, there isn't a `package.json`, so there is nothing for NPM to install or update.

Reply:
Does `npm update` do something in your environment?

What I meant by the previous comment is that I think `npm update` does nothing at all, because there is no configuration file for it to use for installation.

Reply:
yes, fixed...